scripts/add_existing_baseyear.py

```diff
@@ -434,12 +443,22 @@ def add_heating_capacities_installed_before_baseyear(
         else:
             efficiency = costs.at[costs_name, "efficiency"]
 
-        for i, grouping_year in enumerate(grouping_years):
-            if int(grouping_year) + default_lifetime <= int(baseyear):
-                continue
-
-            # installation is assumed to be linear for the past default_lifetime years
-            ratio = (int(grouping_year) - int(grouping_years[i - 1])) / default_lifetime
+        valid_grouping_years = pd.Series(
+            [
+                int(grouping_year)
+                for grouping_year in grouping_years
+                if int(grouping_year) + default_lifetime > int(baseyear)
+                and int(grouping_year) < int(baseyear)
+            ]
+        )
+        
+        # get number of years of each interval
+        _years = (valid_grouping_years.diff().shift(-1)
+                  .fillna(baseyear-valid_grouping_years.iloc[-1]))
+        # Installation is assumed to be linear for the past
+        ratios = _years / _years.sum()
```

Comment:
I think this definition is correct but changes compared to the previous definition of ratio using the lifetime
`ratio = (int(grouping_year) - int(grouping_years[i - 1])) / default_lifetime`

Reply:
i agree that this is correct. with the old definition only 95% of existing heating was added, which can be seen form the prenetwork-brownfield